test/insert.js

```diff
@@ -0,0 +1,49 @@
+var Pipeline = require( '..' )
+var assert = require( 'assert' )
+var fs = require( 'fs' )
+var path = require( 'path' )
+var Stream = require( 'stream' )
+
+describe( 'Pipeline.insert()', function() {
+
+  it( 'can insert a stream at specific indices', function( done ) {
```

Comment:
As `.insert()` uses `.splice()` internally, and tests for the pipe order have been added in #7, this should be covered now.